traffic_ops/traffic_ops_golang/iso/iso.go

```diff
@@ -0,0 +1,368 @@
+		statusCode := http.StatusInternalServerError
+		userErr := errors.New("unable to create kickstarter files")
+		api.HandleErr(w, req, tx.Tx, statusCode, userErr, fmt.Errorf("%v: %v", userErr, err))
+		return
+	}
+
+	isoFilename := fmt.Sprintf("%s-%s.iso", ir.fqdn(), ir.OSVersionDir)
+	// strings.ReplaceAll was added in Go 1.12
+	isoFilename = strings.Replace(isoFilename, "/", "_", -1)
+
+	w.Header().Set(httpHeaderContentDisposition, fmt.Sprintf("attachment; filename=%q", isoFilename))
+	w.Header().Set(httpHeaderContentType, httpHeaderContentDownload)
+
+	if err = genISOCmd.stream(w); err != nil {
+		statusCode := http.StatusInternalServerError
+		userErr := errors.New("unable to generate ISO")
+		api.HandleErr(w, req, tx.Tx, statusCode, userErr, fmt.Errorf("%v: %v", userErr, err))
+		return
+	}
+
+	// Create changelog entry
+	err = api.CreateChangeLogBuildMsg(
+		api.ApiChange,
+		api.Created,
+		user,
+		tx.Tx,
+		"ISO",
+		ir.fqdn(),
+		map[string]interface{}{"OS": ir.OSVersionDir},
+	)
+	if err != nil {
+		// At this point, it's not possible to modify the HTTP response.
+		log.Errorf("error creating changelog entry for ISO creation: %v", err)
+	}
+}
+
+// writeRespErrorAlerts writes to w an Alerts JSON response with
+// an error level. Note: api.WriteRespAlertObj could not be used here
+// because it accepts a single "alert", whereas this response
+// may contain multiple alerts.
+func writeRespErrorAlerts(w http.ResponseWriter, req *http.Request, errMsgs []string) {
+	alerts := tc.CreateAlerts(tc.ErrorLevel, errMsgs...)
+
+	body, err := json.Marshal(alerts)
+	if err != nil {
+		statusCode := http.StatusInternalServerError
+		api.HandleErr(w, req, nil, statusCode, errors.New(http.StatusText(statusCode)), fmt.Errorf("error marshalling JSON: %v", err))
+		return
+	}
+
+	w.Header().Set(rfc.ContentType, rfc.ApplicationJSON)
+	w.WriteHeader(http.StatusBadRequest)
+	w.Write(body)
+}
+
+// isoRequest represents the JSON object clients use to
+// request an ISO be generated.
+type isoRequest struct {
+	OSVersionDir  string          `json:"osversionDir"`
+	HostName      string          `json:"hostName"`
+	DomainName    string          `json:"domainName"`
+	RootPass      string          `json:"rootPass"`
+	DHCP          boolStr         `json:"dhcp"`
+	IPAddr        net.IP          `json:"ipAddress"`
+	IPNetmask     net.IP          `json:"ipNetmask"`
+	IPGateway     net.IP          `json:"ipGateway"`
+	IP6Address    net.IP          `json:"ip6Address"`
+	IP6Gateway    net.IP          `json:"ip6Gateway"`
+	InterfaceName string          `json:"interfaceName"`
+	InterfaceMTU  util.JSONIntStr `json:"interfaceMtu"`
+	Disk          string          `json:"disk"`
+	MgmtIPAddress net.IP          `json:"mgmtIpAddress"`
+	MgmtIPNetmask net.IP          `json:"mgmtIpNetmask"`
+	MgmtIPGateway net.IP          `json:"mgmtIpGateway"`
+	MgmtInterface string          `json:"mgmtInterface"`
+	Stream        boolStr         `json:"stream"`
```

Comment:
Does this struct still need this property?

Reply:
it doesnt I guess our docs point out it isnt supported anymore